read-fonts/src/tables/variations.rs

```diff
@@ -11,6 +11,479 @@ pub struct DeltaSetIndex {
+    /// start of the serialized data.
+    const SHARED_POINT_NUMBERS: u16 = 0x8000;
+
+    /// Mask for the low 12 bits to give the shared tuple records index.
+    const COUNT_MASK: u16 = 0x0FFF;
+
+    pub fn bits(self) -> u16 {
+        self.0
+    }
+
+    /// `true` if any tables reference a shared set of point numbers
+    pub fn shared_point_numbers(self) -> bool {
+        (self.0 & Self::SHARED_POINT_NUMBERS) != 0
+    }
+
+    pub fn count(self) -> u16 {
+        self.0 & Self::COUNT_MASK
+    }
+}
+
+impl types::Scalar for TupleVariationCount {
+    type Raw = <u16 as types::Scalar>::Raw;
+    fn to_raw(self) -> Self::Raw {
+        self.0.to_raw()
+    }
+    fn from_raw(raw: Self::Raw) -> Self {
+        let t = <u16>::from_raw(raw);
+        Self(t)
+    }
+}
+
+impl<'a> TupleVariationHeader<'a> {
+    #[cfg(feature = "traversal")]
+    fn traverse_tuple_index(&self) -> traversal::FieldType<'a> {
+        self.tuple_index().0.into()
+    }
+
+    /// Peak tuple record for this tuple variation table — optional,
+    /// determined by flags in the tupleIndex value.  Note that this
+    /// must always be included in the 'cvar' table.
+    pub fn peak_tuple(&self) -> Option<Tuple<'a>> {
+        self.tuple_index().embedded_peak_tuple().then(|| {
+            let range = self.shape.peak_tuple_byte_range();
+            Tuple {
+                values: self.data.read_array(range).unwrap(),
+            }
+        })
+    }
+
+    /// Intermediate start tuple record for this tuple variation table
+    /// — optional, determined by flags in the tupleIndex value.
+    pub fn intermediate_start_tuple(&self) -> Option<Tuple<'a>> {
+        self.tuple_index().intermediate_region().then(|| {
+            let range = self.shape.intermediate_start_tuple_byte_range();
+            Tuple {
+                values: self.data.read_array(range).unwrap(),
+            }
+        })
+    }
+
+    /// Intermediate end tuple record for this tuple variation table
+    /// — optional, determined by flags in the tupleIndex value.
+    pub fn intermediate_end_tuple(&self) -> Option<Tuple<'a>> {
+        self.tuple_index().intermediate_region().then(|| {
+            let range = self.shape.intermediate_end_tuple_byte_range();
+            Tuple {
+                values: self.data.read_array(range).unwrap(),
+            }
+        })
+    }
+
+    /// Compute the actual length of this table in bytes
+    fn byte_len(&self, axis_count: u16) -> usize {
+        const FIXED_LEN: usize = u16::RAW_BYTE_LEN + TupleIndex::RAW_BYTE_LEN;
+        let tuple_byte_len = F2Dot14::RAW_BYTE_LEN * axis_count as usize;
+        let index = self.tuple_index();
+        FIXED_LEN
+            + index
+                .embedded_peak_tuple()
+                .then_some(tuple_byte_len)
+                .unwrap_or_default()
+            + index
+                .intermediate_region()
+                .then_some(tuple_byte_len)
```

Comment:
```suggestion
                .then_some(tuple_byte_len * 2)
```
Intermediate regions are defined by two tuples. This took me a bit of time to find :)